src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -22,11 +23,54 @@ public static class Storage
             public static readonly InteropDescriptor Find = Register("System.Storage.Find", Storage_Find, 0_01000000, TriggerType.Application, CallFlags.None);
             public static readonly InteropDescriptor Put = Register("System.Storage.Put", Storage_Put, GetStoragePrice, TriggerType.Application, CallFlags.AllowModifyStates);
             public static readonly InteropDescriptor PutEx = Register("System.Storage.PutEx", Storage_PutEx, GetStoragePrice, TriggerType.Application, CallFlags.AllowModifyStates);
-            public static readonly InteropDescriptor Delete = Register("System.Storage.Delete", Storage_Delete, 0_01000000, TriggerType.Application, CallFlags.AllowModifyStates);
+            public static readonly InteropDescriptor Delete = Register("System.Storage.Delete", Storage_Delete, GetDeletePrice, TriggerType.Application, CallFlags.AllowModifyStates);
 
-            private static long GetStoragePrice(EvaluationStack stack)
+            private static long GetDeletePrice(ApplicationEngine engine)
             {
-                return (stack.Peek(1).GetByteLength() + stack.Peek(2).GetByteLength()) * GasPerByte;
+                var stack = engine.CurrentContext.EvaluationStack;
+                var key = stack.Peek(1);
+                if (!(engine.CurrentContext.EvaluationStack.Peek() is InteropInterface _interface))
+                    return 0_01000000;
+                StorageContext context = _interface.GetInterface<StorageContext>();
+                StorageKey skey = new StorageKey
+                {
+                    Id = context.Id,
+                    Key = key.GetSpan().ToArray()
+                };
+                var skeyValue = engine.Snapshot.Storages.TryGet(skey);
+                if (skeyValue == null || skeyValue.Value == null || skeyValue.Value.Length == 0 || engine.HasUpdatedKey(skey))
+                    return 0_01000000;
+                return (skeyValue.Value.Length + key.GetByteLength()) * GasPerReleasedByte;
```

Comment:
I think that the cost is not what the we expect in the following case.

```
db: key -> value0 (100 bytes)
step1: put key -> value1 (100bytes)    // price = 0
step2: delete key                      // price = 0_01000000
```

Reply:
I think we need a fully discuss about thie issue.